[integration-cli] [--net none] Fix TestEventsImagePull

Fix TestEventsImagePull so it will still be a functional test, but will pass when the test container has --net none.

Use the local registry in the container to pull an image.

Error output when tests are run with --net none:

```
--- FAIL: TestEventsImagePull (0.13s)
	docker_cli_events_test.go:188: pulling the hello-world image from has failed: Pulling repository hello-world
		time="2015-02-11T23:19:05Z" level="fatal" msg="Get https://index.docker.io/v1/repositories/library/hello-world/images: dial tcp: lookup index.docker.io on 8.8.8.8:53: dial udp 8.8.8.8:53: network is unreachable" 
		, exit status 1
```

--- a/integration-cli/docker_cli_events_test.go
+++ b/integration-cli/docker_cli_events_test.go
@@ -149,6 +149,7 @@ func TestEventsImageUntagDelete(t *testing.T) {
 
 func TestEventsImagePull(t *testing.T) {
 	since := daemonTime(t).Unix()
+	testRequires(t, Network)
 
 	defer deleteImages("hello-world")
 

--- a/integration-cli/docker_cli_run_test.go
+++ b/integration-cli/docker_cli_run_test.go
@@ -147,6 +147,7 @@ func TestRunLeakyFileDescriptors(t *testing.T) {
 // it should be possible to lookup Google DNS
 // this will fail when Internet access is unavailable
 func TestRunLookupGoogleDns(t *testing.T) {
+	testRequires(t, Network)
 	defer deleteAllContainers()
 
 	out, _, _, err := runCommandWithStdoutStderr(exec.Command(dockerBinary, "run", "busybox", "nslookup", "google.com"))

--- a/integration-cli/docker_cli_search_test.go
+++ b/integration-cli/docker_cli_search_test.go
@@ -8,6 +8,7 @@ import (
 
 // search for repos named  "registry" on the central registry
 func TestSearchOnCentralRegistry(t *testing.T) {
+	testRequires(t, Network)
 	searchCmd := exec.Command(dockerBinary, "search", "busybox")
 	out, exitCode, err := runCommandWithOutput(searchCmd)
 	if err != nil || exitCode != 0 {

